manager/src/main/java/io/gingersnapproject/database/DatabaseHandler.java

```diff
@@ -48,12 +49,37 @@ void start(@Observes StartupEvent ignore, Config config) {
    }
 
    public void refreshSchema() {
-      tables = configuration.rules().values().stream()
-            .map(rule -> vendor.describeTable(pool, rule.connector().table()).await().indefinitely())
-            .filter(Objects::nonNull)
-            .collect(Collectors.toMap(t -> t.name(), Function.identity()));
-      table2rule = configuration.rules().entrySet().stream()
-            .collect(Collectors.toMap(t -> t.getValue().connector().table(), t -> t.getKey()));
+      for (Map.Entry<String, Rule> entry : configuration.rules().entrySet()) {
+         String ruleName = entry.getKey();
+         Rule rule = entry.getValue();
+         String tableName = rule.connector().table();
```

Comment:
This got me thinking. Seems like the `table` requirement would only make sense if not using `selectStatement` because it could be anything, not bound only to the configured table.

Reply:
The `table` field is only relevant to Eager rules and https://github.com/gingersnap-project/cache-manager/pull/46 should make the distinction clearer.

Reply:
Correct, the problem is various things use it right now, but I would like to make it completely optional. My goal after this was to have it be mutually exclusive between

1. sql-statement (required)
2. key-columns (optional)
or
1. table (required)
2. key-columns (optional)
3. value-columns (optional)